cmd/devicesimulator/main.go

```diff
@@ -167,23 +165,35 @@ func reportVersion(versionFormat *string) error {
 
 func startAgent(ctx context.Context, agent *agent.Agent, log *logrus.Logger, agentInstance int) {
 	activeAgents.Inc()
-	prefix := agent.GetLogPrefix()
-	err := agent.Run(ctx)
-	if err != nil {
-		// agent timeout waiting for enrollment approval
-		if errors.Is(err, wait.ErrWaitTimeout) {
-			log.Errorf("%s: agent timed out: %v", prefix, err)
-		} else if ctx.Err() != nil {
-			// normal teardown
-			log.Infof("%s: agent stopped due to context cancellation.", prefix)
-		} else {
-			log.Fatalf("%s: %v", prefix, err)
+	err := wait.PollImmediateWithContext(ctx, 2*time.Second, 5*time.Minute, func(ctx context.Context) (bool, error) {
+		// timeout after 30s and retry
+		ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
+		defer cancel()
+
+		prefix := agent.GetLogPrefix()
+		if err := agent.Run(ctx); err != nil {
+			// agent timeout waiting for enrollment approval
+			if errors.Is(err, wait.ErrWaitTimeout) {
+				log.Errorf("%s: agent %d timed out: %v", prefix, agentInstance, err)
```

Comment:
Internally how long does it take for the agent to timeout and return this error? Will it ever be hit here now?